ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -15,39 +15,78 @@
 
 package io.confluent.ksql.schema.ksql;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import io.confluent.ksql.name.ColumnName;
 import io.confluent.ksql.schema.ksql.types.SqlType;
 import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.util.KsqlException;
+import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
 
 public final class SystemColumns {
 
   public static final ColumnName ROWKEY_NAME = ColumnName.of("ROWKEY");
-
   public static final ColumnName ROWTIME_NAME = ColumnName.of("ROWTIME");
   public static final SqlType ROWTIME_TYPE = SqlTypes.BIGINT;
 
+  public static final ColumnName ROWOFFSET_NAME = ColumnName.of("ROWOFFSET");
+  public static final SqlType ROWOFFSET_TYPE = SqlTypes.BIGINT;
+
+  public static final ColumnName ROWPARTITION_NAME = ColumnName.of("ROWPARTITION");
+  public static final SqlType ROWPARTITION_TYPE = SqlTypes.INTEGER;
+
   public static final ColumnName WINDOWSTART_NAME = ColumnName.of("WINDOWSTART");
   public static final ColumnName WINDOWEND_NAME = ColumnName.of("WINDOWEND");
   public static final SqlType WINDOWBOUND_TYPE = SqlTypes.BIGINT;
 
-  public static final int LEGACY_PSEUDOCOLUMN_VERSION_NUMBER = 0;
-  public static final int CURRENT_PSEUDOCOLUMN_VERSION_NUMBER = 5;
 
-  private static final Set<ColumnName> PSEUDO_COLUMN_NAMES = ImmutableSet.of(
-      ROWTIME_NAME
-  );
+  public static final int ROWTIME_PSEUDOCOLUMN_VERSION = 0;
+  public static final int ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION = 1;
+
+  public static final int LEGACY_PSEUDOCOLUMN_VERSION_NUMBER = ROWTIME_PSEUDOCOLUMN_VERSION;
+  public static final int CURRENT_PSEUDOCOLUMN_VERSION_NUMBER =
```

Comment:
Setting this constant to 1 rather than 0 is a functional change, right? I thought the goal was to limit this PR to non-functional changes only.

Reply:
Correct, this is a mistake